staging/kos/pkg/apis/network/types.go

```diff
@@ -170,42 +173,37 @@ type NetworkAttachment struct {
 // - are in the same Kubernetes API namespace.
 type SubnetSpec struct {
 	// IPv4 is the CIDR notation for the v4 address range of this subnet.
+	// It is immutable: attempts to update it will fail.
 	IPv4 string
 
 	// VNI identifies the virtual network.
 	// Valid values are in the range [1,2097151].
+	// It is immutable: attempts to update it will fail.
 	VNI uint32
 }
 
 type SubnetStatus struct {
 	// Validated tells users and consumers whether the subnet spec has passed
-	// validation or not. The fields that undergo validation are VNI and CIDR.
-	// If Validated is true it is guaranteed to stay true until either one of
-	// SubnetSpec.IPv4 or SubnetSpec.VNI is updated.
-	// If Validated is false or unset, there are three possible reasons:
+	// validation or not. The fields that undergo validation are spec.ipv4 and
+	// spec.vni. If Validated is true it is guaranteed to stay true for the
+	// whole lifetime of the subnet. If Validated is false or unset, there are
+	// three possible reasons:
 	// 	(1) Validation has not been performed yet.
 	// 	(2) The subnet CIDR overlaps with the CIDR of another subnet with the
 	//		same VNI.
 	//	(3) The subnet Namespace is different than that of another subnet with
 	// 		the same VNI.
+	// If for a subnet X Validated is false because of other conflicting
+	// subnets, deletion of the conflicting subnets will cause a transition to
+	// true.
 	// +optional
 	Validated bool
 
-	// +optional
-	Errors SubnetErrors
-}
-
-type SubnetErrors struct {
```

Comment:
Unrelated to immutability: we had a SubnetErrors type to wrap and distinguish errors from IPAM and subnet validator. We agreed that having the IPAM write errors in subnet status is not worth it. Thus the SubnetErrors type is removed and replaced by a slice of strings renamed Errors: there's no ambiguity on who populates it, it's always the validator. We can give it a more descriptive names if you want, such as `validatorErrors`, but I think it is ok to leave things unchanged.